src/main/java/io/curity/identityserver/plugin/aws/authentication/AWSAuthenticatorRequestHandler.java

```diff
@@ -16,73 +16,104 @@
+import java.util.Collections;
+import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Optional;
+import java.util.UUID;
 
-import static io.curity.identityserver.plugin.authentication.Constants.Params.PARAM_REDIRECT_URI;
+import static io.curity.identityserver.plugin.aws.descriptor.AWSAuthenticatorPluginDescriptor.CALLBACK;
 
-public class AWSAuthenticatorRequestHandler implements AuthenticatorRequestHandler<RequestModel> {
+public class AWSAuthenticatorRequestHandler implements AuthenticatorRequestHandler<Request>
+{
     private static final Logger _logger = LoggerFactory.getLogger(AWSAuthenticatorRequestHandler.class);
 
     private final AWSAuthenticatorPluginConfig _config;
-    private final OAuthClient _oauthClient;
+    private final AuthenticatorInformationProvider _authenticatorInformationProvider;
+    private final ExceptionFactory _exceptionFactory;
 
-    public AWSAuthenticatorRequestHandler(AWSAuthenticatorPluginConfig config,
-                                          ExceptionFactory exceptionFactory,
-                                          Json json,
-                                          AuthenticatorInformationProvider provider) {
+    public AWSAuthenticatorRequestHandler(AWSAuthenticatorPluginConfig config)
+    {
         _config = config;
-        _oauthClient = new DefaultOAuthClient(exceptionFactory, provider, json, config.getSessionManager());
+        _exceptionFactory = config.getExceptionFactory();
+        _authenticatorInformationProvider = config.getAuthenticatorInformationProvider();
     }
 
     @Override
-    public Optional<AuthenticationResult> get(RequestModel requestModel, Response response) {
-        _logger.info("GET request received for authentication authentication");
-
-        _oauthClient.setServiceProviderId(requestModel.getRequest());
-        return requestAuthentication(response, ImmutableMap.of(PARAM_REDIRECT_URI, _oauthClient.getCallbackUrl()));
+    public Optional<AuthenticationResult> get(Request request, Response response)
+    {
+        _logger.debug("GET request received for authentication authentication");
+
+        String redirectUri = createRedirectUri();
+        String state = UUID.randomUUID().toString();
+        Map<String, Collection<String>> queryStringArguments = new LinkedHashMap<>(5);
+
+        _config.getSessionManager().put(Attribute.of("state", state));
+
+        queryStringArguments.put("client_id", Collections.singleton(_config.getClientId()));
+        queryStringArguments.put("redirect_uri", Collections.singleton(redirectUri));
+        queryStringArguments.put("state", Collections.singleton(state));
+        queryStringArguments.put("response_type", Collections.singleton("code"));
+
+        String scopes = _config.getScope();
+        if (!scopes.contains("openid"))
+        {
+            scopes += " openid";
+        }
+        if (!scopes.contains("profile"))
+        {
+            scopes += " profile";
```

Comment:
we always send `openid` and `profile` no matter what?

Reply:
yes, 
An ID token is only returned if openid scope is requested. The access token can be only used against Amazon Cognito User Pools if aws.cognito.signin.user.admin scope is requested. The phone, email, and profile scopes can only be requested if openid scope is also requested. These scopes dictate the claims that go inside the ID token.